pkg/okta/okta.go

```diff
@@ -29,23 +33,39 @@ func (f oktaMiddlewareFactory) jwt(logger *zap.Logger, authHeader string) (*jwtv
 	return f.verifier.VerifyAccessToken(bearerToken)
 }
 
-func (f oktaMiddlewareFactory) newUser(logger *zap.Logger, jwt *jwtverifier.Jwt) (models.User, error) {
+func (f oktaMiddlewareFactory) newPrincipal(jwt *jwtverifier.Jwt) (*authn.EUAPrincipal, error) {
 	euaID := jwt.Claims["sub"].(string)
 	if euaID == "" {
-		return models.User{}, errors.New("unable to retrieve EUA ID from JWT")
+		return nil, errors.New("unable to retrieve EUA ID from JWT")
+	}
+
+	// the current assumption is that anyone with an appropriate
+	// JWT provided by Okta for EASi is allowed to use EASi
+	// as a viewer/submitter
+	jcEASi := true
+
+	// need to check the claims for empowerment as a reviewer
```

Comment:
It might be worth putting this block into a well-named helper function, maybe something like this:

```go
jwtHasGRT := jwtContainsGroupClaim(jwt, f.codeGRT)
```

As a reader, I spend a lot of time looking at this block of code - but the particulars of the implementation are not helpful when I'm trying to understand what `newPrincipal` does at a high level.

Reply:
Good idea! Also, this lets me unit test it a bit better too!